### PR TITLE
feat: complete all remaining improvement plan tasks

### PR DESCRIPTION
**Pull request by luandro**
## Summary

Completes all 14 remaining tasks from the improvement plan (`plans/2026-04-18-2026-04-18-awana-labs-improvements-v5.md`), bringing the total to **39/39 items done**.

### Code Quality
- **3.5** Remove unused `parseProject()` function from `src/types/project.schema.ts`

### Test Coverage (+110 new tests)
- **7.1.1** `src/lib/sort-projects.test.ts` — 13 tests (usage/state/date ordering, edge cases)
- **7.1.2** `src/lib/issue-parser.test.ts` — 58 tests (all 8 exported functions)
- **7.2.1** `src/lib/status-utils.test.ts` — 10 tests (status colors, translation keys)
- **7.2.2** `src/components/ScrollToTop.test.tsx` — 5 tests (scroll visibility, click)
- **7.2.3** `src/components/Header.test.tsx` — 7 tests (scroll state, logo, navigation)
- **7.2.4** `src/components/ErrorBoundary.test.tsx` — 7 tests (error catching, retry, i18n)
- **7.2.5** `src/lib/utils.test.ts` — 10 tests (formatRelativeTime all ranges)

### Architecture
- **4.1** Consolidate language detection — disabled i18next detector caches, `awana-labs-language` is now the single source of truth
- **4.2** Evaluate `@octokit/rest` — already isolated in own chunk (19.85 KB gzip), kept as-is
- **4.3** Functional ErrorBoundary — extracted `ErrorDisplay` with `useTranslation()`, removed manual `i18n.on/off`

### Dark Mode Foundation
- **5.2** Added `.dark` CSS custom properties (inverted HSL), replaced hardcoded `green`/`amber`/`red` with CSS variable tokens, updated `color-scheme` meta to `"light dark"`, added `darkMode: ["class"]` to Tailwind config

### DevTools
- **6.1** Installed `@tanstack/react-query-devtools` as devDependency, gated behind `import.meta.env.DEV`

### PWA
- **8.1** Generated 192x192 and 512x512 PNG icons from SVG, added to `manifest.json`

## Verification

- `npm run lint` — 0 errors
- `npm run typecheck` — 0 errors
- `npm run test` — 278 tests passing (was 168)
- `npm run build` — production build succeeds
- Pre-commit (lint-staged) and pre-push (typecheck) hooks pass

## Test plan

- [ ] Verify dev mode renders correctly (no visual regressions)
- [ ] Verify language switching persists across page reloads
- [ ] Verify React Query DevTools panel appears in dev mode
- [ ] Verify dark mode activates when `.dark` class is added to `<html>`
- [ ] Run Lighthouse audit to verify no performance regression

<h3>Greptile Summary</h3>

Large improvement sprint completing 39/39 tasks: dark-mode CSS foundation, a new `useTheme` hook with FOUC prevention, `ThemeToggle` component, `ErrorBoundary` refactor to functional `ErrorDisplay`, i18n language-detection consolidation, React Query DevTools gate, PWA icons, and 110 new tests.

- **Dark mode**: `useTheme` manages light/dark/system preference via `useSyncExternalStore`; an inline FOUC script in `index.html` applies the `dark` class before React hydrates; CSS custom properties in `index.css` cover all design tokens for both modes.
- **ErrorBoundary**: replaced manual `i18n.on/off` with a functional `ErrorDisplay` child that uses `useTranslation()`; adds per-error UUID for observability.
- **i18n**: `i18next` detector caches disabled, `awana-labs-language` is now the sole write path; a one-time migration effect in `LanguageProvider` copies `i18nextLng` → `awana-labs-language` on first boot.

<h3>Confidence Score: 5/5</h3>

Safe to merge — all previously flagged issues are addressed and the new code is well-structured with good test coverage.

The theme system, error boundary refactor, i18n consolidation, and FOUC prevention are all implemented correctly. The FOUC script now handles the 'system' stored value, the useEffect dependency includes systemPreference so OS-level changes propagate, and the ErrorBoundary no longer manually subscribes to i18n events. The only finding is a minor ARIA role mismatch on the ThemeToggle that doesn't affect runtime behavior.

src/components/ThemeToggle.tsx — role="switch" semantics don't match the three-state cycle.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src/components/ThemeToggle.tsx | Three-state toggle (light/dark/system) uses role="switch" (binary ARIA semantics) — screen readers cannot express the "system" state via aria-checked. |
| src/hooks/useTheme.ts | New hook using useSyncExternalStore for theme persistence; correctly handles system-preference changes via dedicated subscription and effect. |
| src/components/ErrorBoundary.tsx | Refactored to delegate rendering to functional ErrorDisplay; adds per-error UUID, removes manual i18n event subscriptions, error message shown only in DEV. |
| src/lib/i18n.ts | Removed detector caches and narrowed detection order; lookupLocalStorage now targets awana-labs-language to consolidate language source of truth. |
| src/hooks/useLanguage.tsx | Adds one-time legacy key migration (i18nextLng → awana-labs-language) and first-visit persistence effect; effects run in declaration order so migration fires before persistence check. |
| index.html | FOUC-prevention script correctly handles light/dark/system stored values before React hydrates; color-scheme meta updated to "light dark". |
| src/index.css | Added .dark CSS custom properties covering all design tokens; status color tokens replace hardcoded green/amber/red values. |
| src/App.tsx | ReactQueryDevtools conditionally lazy-loaded inside QueryClientProvider, gated on import.meta.env.DEV — correctly excluded from production bundle. |
| src/lib/utils.ts | formatRelativeTime now uses raw diffDays (not abs) to distinguish past/future; future years use Intl.RelativeTimeFormat, past years use custom translation key. |
| src/lib/status-utils.ts | Status badge classes migrated from hardcoded Tailwind color names to CSS variable tokens; supports dark mode via .dark overrides in index.css. |
| public/manifest.json | Added 192x192 and 512x512 PNG icons with "purpose": "any maskable" for PWA adaptive icon support. |

</details>

</details>

<details open><summary><h3>Comments Outside Diff (1)</h3></summary>

1. `index.html`, line 20-25 ([link](https://github.com/digidem/awana-labs/blob/d78f73301588c02e9fff18250b7596c48084bafb/index.html#L20-L25)) 

   <a href="#">P1</a> **FOUC script misses `"system"` theme value**

   `useTheme.ts` stores one of three values: `"light"`, `"dark"`, or `"system"`. When a user has explicitly chosen System mode (`localStorage.getItem('awana-labs-theme') === 'system'`), `stored` is the truthy string `'system'`. The condition `!stored` is therefore `false`, so the dark class is never applied even when `prefersDark` is `true` — causing a flash of the light theme on every page load.

   

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: index.html
   Line: 20-25

   Comment:
   **FOUC script misses `"system"` theme value**

   `useTheme.ts` stores one of three values: `"light"`, `"dark"`, or `"system"`. When a user has explicitly chosen System mode (`localStorage.getItem('awana-labs-theme') === 'system'`), `stored` is the truthy string `'system'`. The condition `!stored` is therefore `false`, so the dark class is never applied even when `prefersDark` is `true` — causing a flash of the light theme on every page load.

   

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>
</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 1 code review issue. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 1
src/components/ThemeToggle.tsx:31-35
The ARIA `role="switch"` represents a binary on/off control, but this button cycles through three states: light, dark, and system. `aria-checked` is boolean, so screen readers can never announce the "system" mode — they'll only hear "on" or "off". A plain `role="button"` with the existing descriptive `aria-label` (which already names the current and next state) is both accurate and sufficient.

```suggestion
    <button
      onClick={() => setTheme(nextTheme)}
      className="inline-flex items-center justify-center p-2 rounded-lg text-foreground/70 hover:text-foreground hover:bg-accent transition-colors"
      type="button"
```


`````

</details>

<sub>Reviews (27): Last reviewed commit: ["fix(e2e): resolve 5 failing tests on CI"](https://github.com/digidem/awana-labs/commit/5f0059bad5c76c8d0880a96ef714855e4dcde960) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=28931915)</sub>

### CODE REVIEW

File: src/components/ErrorBoundary.test.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+const { mockI18n } = vi.hoisted(() => ({
+  mockI18n: {
+    t: vi.fn((key: string) => key),
+    on: vi.fn(),
+    off: vi.fn(),
+  },
+}));
+
+vi.mock("@/lib/i18n", () => ({ default: mockI18n }));
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/components/ErrorBoundary.test.tsx]: Fixed in b2f5e89: replaced stale vi.mock('@/lib/i18n') with vi.mock('react-i18next') to match the component's actual import.

File: src/components/ErrorBoundary.tsx [hunk 2]
```diff
@@ -10,6 +10,44 @@ interface ErrorBoundaryState {
...
+ * Functional error display component with i18n support.
+ * Extracted from the class boundary so translations update reactively
+ * without manual i18n.on("languageChanged") + forceUpdate().
+ */
+function ErrorDisplay({
+  onRetry,
+}: {
+  error: Error | null;
+  onRetry: () => void;
+}) {
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/components/ErrorBoundary.tsx:27]: Fixed in b2f5e89: ErrorDisplay now renders error?.message conditionally in a styled paragraph. Added test coverage for the empty-message case.

File: public/manifest.json [hunk 3]
```diff
@@ -11,6 +11,16 @@
...
+    {
+      "src": "/awana-labs/icon-192.png",
+      "type": "image/png",
+      "sizes": "192x192"
+    },
+    {
+      "src": "/awana-labs/icon-512.png",
+      "type": "image/png",
+      "sizes": "512x512"
     }
```
[the remark this answers is not in the document]
Comment (luandro) [reply public/manifest.json:26]: Fixed in b2f5e89: added "purpose": "any maskable" to both PNG icon entries in manifest.json.

File: src/components/ErrorBoundary.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (luandro) [reply src/components/ErrorBoundary.test.tsx]: Fixed in b2f5e89: replaced stale vi.mock('@/lib/i18n') with vi.mock('react-i18next') to match the component's actual import.

File: src/components/ErrorBoundary.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (luandro) [reply src/components/ErrorBoundary.tsx:27]: Fixed in b2f5e89: ErrorDisplay now renders error?.message conditionally in a styled paragraph. Added test coverage for the empty-message case.

File: public/manifest.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (luandro) [reply public/manifest.json:26]: Fixed in b2f5e89: added "purpose": "any maskable" to both PNG icon entries in manifest.json.

File: PROGRESS.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
+# Awana Labs Improvement Plan — Progress Tracker
+
+> Auto-generated from plan `plans/2026-04-18-2026-04-18-awana-labs-improvements-v5.md`
```
[the remark this answers is not in the document]
Comment (luandro) [reply PROGRESS.md]: Fixed in c121ebf: removed unverifiable plan path reference from PROGRESS.md banner.

File: src/components/ErrorBoundary.tsx [hunk 5]
```diff
@@ -10,6 +10,50 @@ interface ErrorBoundaryState {
...
+  return (
+    <div className="min-h-screen flex items-center justify-center bg-background px-6">
+      <div className="max-w-md text-center">
+        <h1 className="text-2xl font-bold text-destructive mb-4">
+          {t("errorBoundary.title")}
+        </h1>
+        <p className="text-muted-foreground mb-6">
+          {t("errorBoundary.description")}
+        </p>
+        {error?.message && (
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/components/ErrorBoundary.tsx]: Fixed in c121ebf: gated error.message display behind import.meta.env.DEV so raw error text only appears in development.

File: src/lib/i18n.ts [hunk 6]
```diff
@@ -37,22 +37,15 @@ i18n
...
-        "localStorage",
-        "sessionStorage",
-        "navigator",
-        "htmlTag",
-      ],
+      // Detection order for language resolution.
+      // localStorage/sessionStorage are excluded from *caches* so that the
+      // custom LanguageProvider (which uses key "awana-labs-language") is the
+      // single source of truth for persisted language preferences.
+      order: ["querystring", "navigator", "htmlTag"],
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/lib/i18n.ts]: Fixed in c121ebf: added localStorage to i18n detection order with lookupLocalStorage key 'awana-labs-language', so the persisted preference is read on init before React mounts. caches remains empty so LanguageProvider stays the single source of truth for writes.

File: index.html [hunk 7]
```diff
@@ -3,7 +3,7 @@
   <head>
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <meta name="color-scheme" content="only light" />
+    <meta name="color-scheme" content="light dark" />
```
[the remark this answers is not in the document]
Comment (luandro) [reply index.html:6]: Fixed in c121ebf: changed color-scheme meta to 'light' only since there is no runtime code applying .dark class from system preference.

File: PROGRESS.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (luandro) [reply PROGRESS.md]: Fixed in c121ebf: removed unverifiable plan path reference from PROGRESS.md banner.

File: src/components/ErrorBoundary.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (luandro) [reply src/components/ErrorBoundary.tsx]: Fixed in c121ebf: gated error.message display behind `import.meta.env.DEV` so raw error text only appears in development. Error is still logged via console.error in componentDidCatch.

File: src/lib/i18n.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (luandro) [reply src/lib/i18n.ts]: Fixed in c121ebf: added localStorage to i18n detection order with `lookupLocalStorage: 'awana-labs-language'`, so the persisted preference is read on init before React mounts. `caches` remains empty so LanguageProvider stays the single source of truth for writes.

File: index.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (luandro) [reply index.html:6]: Fixed in c121ebf: changed color-scheme meta to `light` only since there is no runtime code applying `.dark` class from system preference.

File: src/lib/utils.test.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    expect(result).toContain("timeAgo.overYears");
+    expect(result).toContain("3");
+  });
+
+  it("works with different locales", () => {
+    expect(formatRelativeTime(daysAgo(0), "pt", t)).toBe("hoje");
+    expect(formatRelativeTime(daysAgo(1), "es", t)).toBe("ayer");
+  });
+
+  it("handles future dates", () => {
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/lib/utils.test.ts:70]: Addressed: added yearsFromNow helper and test case 'handles future dates beyond 12 months (overYears branch)' in src/lib/utils.test.ts to cover the 12+ months future-date path. Also fixed color-scheme meta tag (light dark) and converted ReactQueryDevtools to dynamic import.

File: src/lib/i18n.ts [hunk 9]
```diff
@@ -37,22 +37,17 @@ i18n
...
+      // Key used by LanguageProvider to persist language preference
+      lookupLocalStorage: "awana-labs-language",
       lookupQuerystring: "lng",
-      lookupCookie: "i18next",
-      lookupLocalStorage: "i18nextLng",
-      lookupSessionStorage: "i18nextLng",
-      // Cache user language selection
-      caches: ["localStorage", "cookie"],
+      // Do NOT cache — LanguageProvider manages persistence via "awana-labs-language"
+      caches: [],
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/lib/i18n.ts:50]: Fixed in e321a79: added mount-only effect in LanguageProvider that persists the i18n-detected language to localStorage when storage is empty, so first-visit detection via ?lng= or navigator.language survives page reload.

File: src/components/Header.test.tsx [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+  ),
+}));
+
+describe("Header", () => {
+  let scrollCallback: (scrollY: number) => void;
+  let heroElement: HTMLElement;
+  let mockH1: HTMLElement;
+
+  beforeEach(() => {
+    vi.mocked(useScrollListener).mockImplementation((cb) => {
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/components/Header.test.tsx:30]: Fixed in 3dbf7ea: added explicit act(() => scrollCallback(window.scrollY)) calls after render() in tests verifying initial mounted state, plus new tests for mid-page refresh scenarios in both Header and ScrollToTop.

File: src/components/ErrorBoundary.test.tsx [hunk 11]
```diff
@@ -0,0 +1,160 @@
+import { describe, it, expect, vi, beforeEach, afterEach } from "vitest";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { ErrorBoundary } from "@/components/ErrorBoundary";
+
+vi.mock("react-i18next", () => ({
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/components/ErrorBoundary.test.tsx:5]: Fixed in ef4112f: replaced ineffective placeholder tests with real i18n reactivity tests in new ErrorBoundary.i18n.test.tsx using real i18n + I18nextProvider. Verifies error UI updates from English to Portuguese when i18n.changeLanguage() fires.

File: src/lib/utils.test.ts [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+    expect(formatRelativeTime(daysFromNow(1), "en", t)).toBe("tomorrow");
+    expect(formatRelativeTime(daysFromNow(3), "en", t)).toBe("in 3 days");
+  });
+
+  it("handles future dates beyond 12 months (overYears branch)", () => {
+    const result = formatRelativeTime(yearsFromNow(2), "en", t);
+    expect(result).toContain("timeAgo.overYears");
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/lib/utils.test.ts:87]: Fixed in 4ad41a7: improved test assertion to use exact match and added comment documenting the known past-tense limitation for future dates in the overYears branch.

File: src/lib/i18n.test.ts [hunk 13]
```diff
@@ -154,5 +153,50 @@ describe("i18n configuration", () => {
...
+        : detectedLanguage;
+      expect(resolvedLanguage).toBe("pt");
+
+      // Simulate what LanguageProvider does on mount: if localStorage is empty,
+      // persist the detected language so it survives a page reload.
+      const storageKey = "awana-labs-language";
+      expect(localStorage.getItem(storageKey)).toBeNull();
+
+      // Apply the same logic as the mount effect in LanguageProvider
+      if (localStorage.getItem(storageKey) === null && resolvedLanguage) {
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/lib/i18n.test.ts]: Fixed in 4ad41a7: rewrote test to render the real LanguageProvider component via renderHook instead of duplicating the mount-effect logic inline.

File: src/hooks/useTheme.ts [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+    listeners.forEach((fn) => fn());
+  }, []);
+
+  const resolvedTheme: "light" | "dark" =
+    theme === "system" ? systemPreference : theme;
+
+  // Re-apply on mount to stay in sync
+  useEffect(() => {
+    applyTheme(theme);
+  }, [theme]);
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/hooks/useTheme.ts]: Fixed in fc8b849: changed the useEffect dependency from `theme` to `resolvedTheme` so that DOM updates when system color-scheme preference changes while in system mode. The effect now directly toggles the `dark` class based on the resolved theme value.

File: src/lib/utils.test.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (luandro) [reply src/lib/utils.test.ts:87]: Already addressed in a prior commit: the test at src/lib/utils.test.ts:75-80 now asserts `result.toBe("in 2 years")` which correctly validates the `rtf.format` path for future dates beyond 12 months. The `overYears` i18n key is only used for past dates (line 56 in utils.ts), while future dates use `rtf.format(diffYears, "year")` (line 54).

File: index.html [hunk 15]
```diff
@@ -46,6 +46,16 @@
   ]
 }
 </script>
+    <script>
+      // Apply dark mode before paint to avoid FOUC
+      (function() {
+        var stored = localStorage.getItem('awana-labs-theme');
+        var prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+        if (stored === 'dark' || (!stored && prefersDark)) {
```
[the remark this answers is not in the document]
Comment (luandro) [reply index.html]: Fixed in 8a7e7e9: the FOUC script now correctly handles the `'system'` stored value by checking `stored === 'system' || !stored` before testing `prefersDark`, matching the runtime resolution logic in useTheme.ts.

**Comment by luandro** [comment]
Addressed remaining review feedback in c1ba968:

- **useTheme.ts**: Wrapped `getThemeSnapshot` and `setTheme` localStorage calls in try-catch, matching the pattern in `useLanguage.tsx`. Theme still applies to DOM when persistence fails.
- **utils.test.ts**: Added boundary tests for past/future 12+ month split logic. Past dates use i18n key `timeAgo.overYears` (not `rtf.format`), future dates use `rtf.format`. New tests verify both paths and the exact 12-month boundary.

All 286 tests pass, typecheck and lint clean.
